### PR TITLE
Create navbar

### LINKED ISSUE #25: Nav Bar: Navigation Bar
Image

### PR DESCRIPTION
**Pull request by Raymond0T**
#### What is the purpose of your change?
-changed routes and added navbar (fixes #25 )

#### What areas of the application does this change impact? (changes API, folder structure, etc)
-front-end and routing

#### What features/fixes are deferred/not handled in this PR?
-

#### Screenshot/link:
image
https://deploy-preview-118--pick-a-major.netlify.app/

## Checklist
- [ ] Adds unit tests
- [ ] Has a work item linked to the PR

### CODE REVIEW

File: tmt-app/.vscode/settings.json
Comment (jwu910) [inline tmt-app/.vscode/settings.json:1]: delete this file, should not be committing files that you do not intend to

**Review by michachan — APPROVED** [review]
Love it

File: .github/workflows/lintchecks.yml
Comment (jwu910) [inline .github/workflows/lintchecks.yml:1]: Can you undo any changes here that aren't needed?

File: tmt-app/.prettierrc.json
Comment (jwu910) [inline tmt-app/.prettierrc.json:1]: Can we undo any unnecessary changes here?

File: tmt-app/src/Components/NavBarWrapper.jsx
Comment (jwu910) [inline tmt-app/src/Components/NavBarWrapper.jsx:1]: Why not just put this outlet in the navbar?

File: tmt-app/src/Components/Team.jsx
Comment (jwu910) [inline tmt-app/src/Components/Team.jsx:1]: Should this go in pages?

**Review by jwu910 — CHANGES_REQUESTED** [review]

File: tmt-app/src/Components/NavBar.jsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                <ListItem sx={{ justifyContent: 'flex-end' }}>
+                  <IconButton
+                    aria-label="options"
+                    color="black"
+                    onClick={handleCloseNavMenu}
+                  >
+                    <CloseIcon />
+                  </IconButton>
+                </ListItem>
+                {tabs.map((tab, id) => {
```
Comment (jwu910) [inline tmt-app/src/Components/NavBar.jsx:94]: map's 2nd param is `index` btw, we should use that over `id` to not get confused

File: tmt-app/src/Components/NavBar.jsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                            sx={{
+                              margin: '0 1rem',
+                              color: 'Boba.main',
+                            }}
+                          >
+                            <Typography variant="h5">{tab.name}</Typography>
+                          </Link>
+                        </ListItemButton>
+                      </ListItem>
+                    </List>
```
Comment (jwu910) [inline tmt-app/src/Components/NavBar.jsx:113]: btw why do we need a whole new list item here for every tab?

File: tmt-app/src/Components/NavBar.jsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          <Box
+            sx={{
+              flexGrow: 1,
+              display: { xs: 'none', md: 'flex' },
+              justifyContent: 'flex-end',
+              alignItems: 'center',
+            }}
+          >
+            {tabs.map((tab, id) => {
+              return id === tabs.length - 1 ? (
```
Comment (jwu910) [inline tmt-app/src/Components/NavBar.jsx:128]: i think the CTA take the quiz is sort of its own action button, and doesn't quite fit in to the concept of tabs here.

Does design intend for the nav bar to have any type of active indicator? 

https://mui.com/material-ui/react-tabs/ 

maybe consider some pre-built functionality here too

File: tmt-app/src/Components/NavBar.jsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                  component={RouterLink}
+                  underline="none"
+                  sx={{
+                    color: 'white',
+                    margin: '0 1rem',
+                    padding: '0.5rem 3rem',
+                    backgroundColor: 'black',
+                    borderRadius: '30px',
+                    ':hover': {
+                      backgroundColor: '#484848',
```
Comment (jwu910) [inline tmt-app/src/Components/NavBar.jsx:141]: We should pull all colors from theme, not hard code them

image
This also doesn't match design spec. can we take a little extra time reviewing the designs as part of the development process?

image

**Review by jwu910 — CHANGES_REQUESTED** [review]

File: .github/workflows/lintchecks.yml
Comment (Raymond0T) [reply .github/workflows/lintchecks.yml:1]: I don't remember touching this file. Could this be a change done automatically by eslint/prettier?

File: tmt-app/.prettierrc.json
Comment (Raymond0T) [reply tmt-app/.prettierrc.json:1]: same thing... I don't remember touching this file. Could this be a change done automatically by eslint/prettier?

File: tmt-app/src/Components/NavBarWrapper.jsx
Comment (Raymond0T) [reply tmt-app/src/Components/NavBarWrapper.jsx:1]: I will try that and make the changes

File: tmt-app/src/Components/Team.jsx
Comment (Raymond0T) [reply tmt-app/src/Components/Team.jsx:1]: what do you mean by pages? Do you mean changing it to TeamPage?

File: tmt-app/src/Components/NavBar.jsx (hunk 1 above)
Comment (Raymond0T) [reply tmt-app/src/Components/NavBar.jsx:94]: sounds good! I will make the change

File: tmt-app/src/Components/NavBar.jsx (hunk 2 above)
Comment (Raymond0T) [reply tmt-app/src/Components/NavBar.jsx:113]: not too sure why I did that haha... I will fix it

File: tmt-app/src/Components/NavBar.jsx (hunk 3 above)
Comment (Raymond0T) [reply tmt-app/src/Components/NavBar.jsx:128]: I will check with the team for any active indicator

File: tmt-app/src/Components/NavBar.jsx (hunk 4 above)
Comment (Raymond0T) [reply tmt-app/src/Components/NavBar.jsx:141]: This does not relatively match the design because the nav bar will be too big if we are going by the spec of 123px for the height. I am also using the default max-width from mui to keep the design in-place. Any screen size bigger than the max-width would just add on more empty space to the left and right side.

File: .github/workflows/lintchecks.yml
Comment (jwu910) [reply .github/workflows/lintchecks.yml:1]: If you opened it up, the line endings could have changed. But if you have a habit of just running "git add ." You're gonna run into this issue. Should always review and know what you're committing.

`Git log --stat` will tell you when the file was committed so you can undo it if you wanted to interactive rebase it or just make a new commit fixing these so they net zero on changes

File: tmt-app/.prettierrc.json
Comment (jwu910) [reply tmt-app/.prettierrc.json:1]: You changed this one trying to find new node versions. You can read this through your commit history

File: tmt-app/src/Components/Team.jsx
Comment (jwu910) [reply tmt-app/src/Components/Team.jsx:1]: Never mind it doesn't seem to be the pattern adopted here

File: .github/workflows/lintchecks.yml
Comment (Raymond0T) [reply .github/workflows/lintchecks.yml:1]: yeah I see it in my git history but shouldn't we leave this because it is fixing the carriage return?

File: tmt-app/.prettierrc.json
Comment (Raymond0T) [reply tmt-app/.prettierrc.json:1]: I believe the file I changed for the new version was .nvmrc. However, I don't see this in my git history.

File: .github/workflows/lintchecks.yml
Comment (jwu910) [reply .github/workflows/lintchecks.yml:1]: I can't see from my phone 💀💀 @michachan ?

Comment (michachan) [reply .github/workflows/lintchecks.yml:1]: Should run the autocrlf command and commit any changed files

File: tmt-app/src/Components/NavBar.jsx (hunk 3 above)
Comment (Brian-Miki) [reply tmt-app/src/Components/NavBar.jsx:128]: @jwu910 @Raymond0T Design says to just keep it as is.

Comment (jwu910) [reply tmt-app/src/Components/NavBar.jsx:128]: Can we spawn a new ticket and raise the accessibility issue and backlog it then?

Visual indicator for active page I thought was pretty standard accessible design

**Comment by michachan** [comment]
Merging to resolve blockers
